include/crm/common/util.h

```diff
@@ -51,6 +51,7 @@ gboolean crm_is_true(const char *s);
 int crm_str_to_boolean(const char *s, int *ret);
 long long crm_parse_ll(const char *text, const char *default_text);
 int crm_parse_int(const char *text, const char *default_text);
+double pcmk_parse_double(const char *text, const char *default_text);
```

Comment:
I could see either:
- keeping scan_double() as a helper function and converting pcmk_parse_double to an internal function that wraps scan_double(), or
- modifying scan_double()'s interface to make it a friendly internal function.

I wouldn't want to use scan_double() as an internal function with its current interface.

The reason is that scan_double() returns an error/success code and stores the parsed value in a pointer-to-double parameter. That's nice if we want to do something like setting the pointer to NULL if the parse fails (which we don't currently do). But it doesn't allow us to use the return value as an argument to another function, as we sometimes do with crm_parse_ll() and crm_parse_int(). To me, that's a handy capability. Otherwise we have to do the parse first, storing the result in some variable, and pass that variable to another function.

You also said in a different comment:
> going forward for new functions we only need to use the standard return code.

For the reasons given above (ease of passing the return value as a function argument), I think we should either:
- continue setting errno, or
- use a status argument similar to what waitpid() uses, to store the standard Pacemaker return code.

I lean toward the latter. I'll probably do something like that in the next push. Let me know what you think :)

Reply:
> Otherwise we have to do the parse first, storing the result in some variable, and pass that variable to another function.

Actually I prefer that -- implicitly handling errors inside a function and having the caller ignore it is a recurring source of bugs. I've seen more than one where it didn't matter for a function's initial use case but a later use really needed to be checking for errors, but since the easy interface was available, it was overlooked.

I prefer returning the code, rather than putting the code in an argument or errno, precisely because it does force the developer to think about error checking (as well as for consistency).

The only time I don't mind returning a result directly (instead of a standard code) is when the function can't fail (due to simplicity, or asserting on errors), or the interface is dictated by its usage with an external library (e.g. a glib callback).

Reply:
Ah, understandable concerns. Will be resolved in the next push.